client/src/Components/Register.jsx

```diff
@@ -1,7 +1,201 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { GrLocation } from 'react-icons/gr';
+import { FaRegUser } from 'react-icons/fa';
+import { FaAngleLeft, FaRegEnvelope } from 'react-icons/fa6';
+import { FiLock } from 'react-icons/fi';
+import { PiUserListBold } from 'react-icons/pi';
+import { Link, useNavigate } from 'react-router-dom';
+import { fetchRegister } from '../utils/fetchUserData';
+import '../css/LoginSignUp.css';
+
+export default function Register({ setUser }) {
+  const [registerData, setRegisterData] = useState({
+    firstName: '',
+    lastName: '',
+    email: '',
+    username: '',
+    password: '',
+    confirmPassword: '',
+    zipCode: '',
+  });
+  const [validPassword, setValidPassword] = useState(false);
+  const [passwordMatch, setPasswordMatch] = useState(true);
+
+  const navigate = useNavigate();
+
+  const handleRegister = async () => {
+    if (passwordMatch && validPassword) {
+      try {
+        const response = await fetchRegister(registerData);
+        setUser(response.user.displayUsername);
+        navigate('/');
+      } catch (err) {
+        console.error(err);
+      }
+    }
+  };
+
+  const handleChange = (e) => {
+    e.preventDefault();
+    setRegisterData({ ...registerData, [e.target.id]: e.target.value });
+  };
+
+  const handlePasswordCheck = () => {
+    const [pass, confPass] = [registerData.password, registerData.confirmPassword];
+
+    // If passwords don't match, user will be notified.
+    if (confPass.length && pass !== confPass) {
+      setPasswordMatch(false);
+    } else if (pass === confPass) {
+      setPasswordMatch(true);
+    }
+
+    // If password is shorter than 6 characters, user will be notified.
+    if (pass.length && pass.length >= 6) {
```

Comment:
In the input there is set to 4 & here is set to 6, lets define a const at the top for both to be set by. Is this the only form of validation we have on the password entry? should add either some other checks in regex but that can be another pr.